### PR TITLE
NewsAPI Code review merge request

### LINKED ISSUE #2: Feature: Frontend - Created Context, linked Axios and made cards to display data
Created _Context_ with **Context API**, linked _axios_ and made cards to display data
Created _Navbar_, div that acts as a wrapper for every page, and linked **Browser Router** to the project for multi-page feature.

### LINKED ISSUE #3: Feature: Search Bar with filtration for title
Added a Search Bar with a filter function that checks if any of crawled titles includes term from search bar in it.

### LINKED ISSUE #4: Feature: Load More Button. Filtering Components moved to different folder due to overcomplicated code
Load More Button. Filtering Components moved to different folder due to overcomplicated code.

### LINKED ISSUE #5: Feature: Input bar with Filtering Chips, each one with its own value
These "Chips", are actually component from Mantine UI, and each one has unique value. They help with sorting, by popularity, relevance, and published date. Input bar is just there to search if any of articles from newsapi contains typed term. Also, I organized main and sub components, in their own folders.

### LINKED ISSUE #6: Removed "inline styles" that "client" hates. Also started working on full page for article.
I started working on full page, but API request limit ran out.

### LINKED ISSUE #8: Full Page Content
Made Full Page with useNavigate and useLocation hooks. Also, moved 90% of the styles to Mantine UI **createStyles**. Styles that are left to move are props of components, that I need to figure out how to move.

### LINKED ISSUE #9: Feature: Toggle Theme

### PR DESCRIPTION
**Pull request by hamzaplojovic**
Done : - Created Context provider with React context API. 
            - Axios get to add data to context. 
            - Articles page
            - Navbar
            - Layout wrapper
            - BrowserRouter

ToDo: - BrowserRouter
           - Searching and Filtering
           - Full Article Page with all data items

### CODE REVIEW

File: src/Components/Layout/Layout.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { NavbarMinimal } from "../Navbar/Navbar";
+
+export function Layout({
+    children,
+}: {
+    children: React.ReactNode | React.ReactNode[];
+}) {
+    return (
+        <div
+            style={{
```
Comment (suadsuljovic) [inline src/Components/Layout/Layout.tsx]: You are using emotion library so all CSS should be created with it.

File: src/Components/Layout/Layout.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                width: "100%",
+                alignItems: "center",
+            }}
+        >
+            <NavbarMinimal />
+            <div
+                style={{
+                    width: "100%",
+                    justifyContent: "center",
+                }}
```
Comment (suadsuljovic) [inline src/Components/Layout/Layout.tsx]: Use emotion instead of inline css

File: src/Components/News/FilteringComponents/Filtering.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    });
+
+    return (
+        <MediaQuery
+            query="(max-width:1000px)"
+            styles={{ flexDirection: "column" }}
+        >
+            <div className={classes.container}>
+                {data.articles &&
+                    data.articles.slice(0, pageSize).map((item: any) => {
```
Comment (suadsuljovic) [inline src/Components/News/FilteringComponents/Filtering.tsx]: Why are you changing the page size here? You can just use the 20 as you get from the API.

File: src/Components/News/FilteringComponents/LoadMore.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        <div className={classes.buttonwrapper}>
+            <Button
+                size="md"
+                onClick={() => {
+                    if (props.data.articles.length > props.state) {
+                        props.setState(props.state + 5);
+                    } else {
+                        alert("No content!");
+                    }
+                }}
```
Comment (suadsuljovic) [inline src/Components/News/FilteringComponents/LoadMore.tsx]: Not sure what state variable means here. If it represents the number of article in states than rename it to match that.

File: src/Components/News/FilteringComponents/Mapping.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            query="(max-width:1000px)"
+            styles={{ flexDirection: "column" }}
+        >
+            <div className={classes.container}>
+                {props.data &&
+                    props.data.articles
+                        .slice(0, props.pageSize)
+                        .map((item: any) => (
+                            <NewsArticle data={item} key={item.url} />
+                        ))}
```
Comment (suadsuljovic) [inline src/Components/News/FilteringComponents/Mapping.tsx]: There is no need to change page size unless you have a really good reason.

File: src/Components/News/FullPageSub/FullPageContent.tsx [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import { Text, Title, Image } from "@mantine/core";
+
+export const FullPageContent = (props: { data: any }) => {
+    console.log(props.data.articles);
```
Comment (suadsuljovic) [inline src/Components/News/FullPageSub/FullPageContent.tsx]: Please remove console.log.

File: src/Components/News/FullPageSub/Wrapper.tsx [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { useParams } from "react-router-dom";
+import { FullPageData } from "../../../context/full";
+import { FullPageContent } from "./FullPageContent";
+
+export const Wrapper = () => {
+    const title: any = useParams().id;
```
Comment (suadsuljovic) [inline src/Components/News/FullPageSub/Wrapper.tsx]: Check how to properly get prop with use Params and how to set the types for it on react cheat sheet website.

File: src/Components/News/News.tsx [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import { useContext, useState } from "react";
+import { NewsContext } from "../../context/context";
+import { Mapping } from "./FilteringComponents/Mapping";
+import { Load } from "./FilteringComponents/LoadMore";
+import { Layout } from "../Layout/Layout";
+
+export const News = () => {
+    const [pageSize, setPageSize] = useState(10);
+    const data: any = useContext(NewsContext);
```
Comment (suadsuljovic) [inline src/Components/News/News.tsx]: This should not be any.

File: src/Components/News/SearchSub/Button.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        justifyContent: "center",
+        marginTop: "3%",
+        padding: "0% 43%",
+    },
+}));
+
+export const SearchButton = (props: {
+    term: string;
+    setClick: React.Dispatch<React.SetStateAction<string>>;
+}) => {
```
Comment (suadsuljovic) [inline src/Components/News/SearchSub/Button.tsx]: Move this to actually type declaration above.

File: src/Components/News/SearchSub/Input.tsx [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+        padding: "0% 20%",
+    },
+}));
+
+export const SearchInput = (props: {
+    term: string;
+    setTerm: React.Dispatch<React.SetStateAction<string>>;
+    search: string;
+    setSearch: React.Dispatch<React.SetStateAction<string>>;
+}) => {
```
Comment (suadsuljovic) [inline src/Components/News/SearchSub/Input.tsx]: Also move it before and add a name to this type.

File: src/context/context.tsx [hunk 11]
```diff
@@ -0,0 +1,27 @@
+import axios from "axios";
+import { createContext, useEffect, useState } from "react";
+
+interface NewsContextInterface {
+    children: React.ReactNode;
+}
+
+export const NewsContext = createContext(undefined);
```
Comment (suadsuljovic) [inline src/context/context.tsx]: You should create type for what this will hold.

File: src/context/context.tsx [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    const apikey = "e75c1dd786c8454d989a14ccd2e3ca27";
+    useEffect(() => {
+        axios
+            .get(
+                `https://newsapi.org/v2/top-headlines?country=us&pageSize=100&apiKey=${apikey}`
+            )
+            .then((r) => {
+                setData(r.data);
+            });
+    }, []);
```
Comment (suadsuljovic) [inline src/context/context.tsx]: I like what you did here. But page size should be actual page size (10 or 20). And when the user clicks you should load more articles with the same page size count.

Comment (suadsuljovic) [reply src/context/context.tsx]: You don't really have infinite scroll implementation here. You are just mocking this with 100 calls.

File: src/context/full.tsx [hunk 13]
```diff
@@ -0,0 +1,17 @@
+import axios from "axios";
+import { useEffect, useState } from "react";
+
+export const FullPageData = (props: { title: string }) => {
```
Comment (suadsuljovic) [inline src/context/full.tsx]: Not sure how you use this. But if this is going to be a hook it should start with use in it's name.

File: src/context/search.tsx [hunk 14]
```diff
@@ -0,0 +1,18 @@
+import axios from "axios";
+import { useEffect, useState } from "react";
+
+export const SearchData = (props: { phrase: string; sortBy: string }) => {
```
Comment (suadsuljovic) [inline src/context/search.tsx]: If you wan't this to be a hook it should start with use keywoard.

**Review by suadsuljovic — CHANGES_REQUESTED** [review]
It looks really good. Great job at doing all of this in such a short time. 
I have left some comments and we can discuss it how to solve it.

**Comment by suadsuljovic** [comment]
Also please keep the PRs smaller next time. 30 files changed is too much

File: src/Components/Layout/Layout.tsx (hunk 1 above)
Comment (hamzaplojovic) [reply src/Components/Layout/Layout.tsx]: I have good excuse for making this mistake. It is much easier to write **inline parameters** that components from Mantine get, but I understand it is really bad for reading the code. I will try to do everything with **styles up**, and just **components in functions**.

File: src/context/context.tsx (hunk 12 above)
Comment (hamzaplojovic) [reply src/context/context.tsx]: I was hoping you are not going to notice it. But it is really sneaky, right?

**Pull request merged**